packages/vector_graphics_compiler/lib/src/svg/masking_optimizer.dart

```diff
@@ -0,0 +1,362 @@
+import 'dart:core';
+import 'dart:typed_data';
+
+import 'package:vector_graphics_compiler/src/svg/node.dart';
+import 'package:vector_graphics_compiler/src/svg/resolver.dart';
+import 'package:vector_graphics_compiler/src/svg/visitor.dart';
+import 'package:vector_graphics_compiler/vector_graphics_compiler.dart';
+import 'package:vector_graphics_compiler/src/svg/path_ops.dart' as path_ops;
+
+class _EmptyNode extends Node {
+  const _EmptyNode();
+
+  @override
+  S accept<S, V>(Visitor<S, V> visitor, V data) {
+    return visitor.visitEmptyNode(this, data);
+  }
+
+  @override
+  void visitChildren(NodeCallback visitor) {}
+}
+
+class _Result {
+  _Result(this.node);
+
+  final Node node;
+  final List<Node> children = <Node>[];
+  Node parent = const _EmptyNode();
```

Comment:
Can you use `Node.empty` instead of adding a new EmptyNode class?

```suggestion
  Node parent = Node.empty
```

Reply:
fixed.